sources/osgStats/glStats.js

```diff
@@ -0,0 +1,110 @@
+var WebGLRenderingContext = window.WebGLRenderingContext;
+var WebGL2RenderingContext = window.WebGL2RenderingContext;
+
+var contextList = [WebGLRenderingContext, WebGL2RenderingContext];
+
+var glProxy = function(f, c) {
+    return function() {
+        c.apply(this, arguments);
+        f.apply(this, arguments);
+    };
+};
+
+var totalDrawArraysCalls = 0,
+    totalDrawElementsCalls = 0,
+    totalUseProgramCalls = 0,
+    totalFaces = 0,
+    totalVertices = 0,
+    totalPoints = 0,
+    totalBindTexures = 0;
+
+for (var i = 0; i < contextList.length; i++) {
+    var WebGLContext = contextList[i];
+    if (!WebGLContext) continue;
+
+    WebGLContext.prototype.drawArrays = glProxy(WebGLContext.prototype.drawArrays, function() {
+        totalDrawArraysCalls++;
+        if (arguments[0] === this.POINTS) totalPoints += arguments[2];
+        else totalVertices += arguments[2];
+    });
+    WebGLContext.prototype.drawElements = glProxy(WebGLContext.prototype.drawElements, function() {
+        totalDrawElementsCalls++;
+        totalFaces += arguments[1] / 3;
```

Comment:
This code is being called also when no stats in the viewer. Not very important but could be nice to have a way to get the "real" number of faces, as this is an approximation.